solr/views.py

```diff
@@ -75,21 +75,33 @@ def cleanup_solr_request(self, payload, user_id='default'):
         :param payload: raw request payload
         :return: sanitized payload
         """
+        def safe_int(val, default=0):
+            try:
+                return int(val)
+            except (ValueError, TypeError):
+                return default
+
         payload['wt'] = 'json'
         max_rows = current_app.config.get('SOLR_SERVICE_MAX_ROWS', 100)
         max_rows *= int(
             request.headers.get('X-Adsws-Ratelimit-Level', 1)
         )
-        if 'rows' in payload and int(payload['rows'][0]) > max_rows:
+
+        # Ensure all values are strings and not lists
+        for key, value in payload.iteritems():
```

Comment:
a problem with this approach is 'fq' - solr can accept multiple filters

Reply:
The code will convert:

```
{
 'fq': [u'{!bitset}'],
 'rows': [u'25'],
 'sort': [u'date desc, bibcode desc'],
 'start': [u'0'],
 'wt': 'json'
}
```

into:

```
{
 'fq': u'{!bitset}',
 'rows': u'25',
 'sort': u'date desc, bibcode desc',
 'start': u'0',
 'wt': 'json'
}
```

In case there are multiple fq, the payload will look like this:

```
{
 'fq': [u'{!bitset}', u'author', u'title'],
 'rows': [u'25'],
 'sort': [u'date desc, bibcode desc'],
 'start': [u'0'],
 'wt': 'json'
}
```

and it will be converted to:

```
{
 'fq': u'{!bitset},author,title',
 'rows': u'25',
 'sort': u'date desc, bibcode desc',
 'start': u'0',
 'wt': 'json'
}
```

Am I misunderstanding your comment? It does not seem to be a problem with multiple fq. Notice also that when the clean function is called:

```
self.cleanup_solr_request(dict(request.args)...)
```

The dict conversion of the argument converts multiple fq (e.g., "URL?fq=author&fq=title") into a single key (i.e. 'fq') with a list of values (i.e. ["author", "title"]).

Reply:
fq needs to be a list of values, note that a single 'fq' conditin can contain a comma - so:

`payload['fq'] = ','.join(['pos(1,author:foo)', 'title'])`

will not give you the right data when you do `payload['fq'].split(',')` -- and if we did that (I didn't look), it would be at least weird - to join array into a string, in order to create a list out of the string in the end...